pkg/rpc/context.go

```diff
@@ -114,29 +114,25 @@ func spanInclusionFuncForClient(
 
 type serverOpts struct {
 	interceptor func(fullMethod string) error
-	tenant      bool
 }
 
 // ServerOption is a configuration option passed to NewServer.
 type ServerOption func(*serverOpts)
 
-// ForTenant is an option to NewServer that results in the server being set
-// up to validate incoming tenants. With this option the server still uses
-// KV-internal node certificates but listens on a dedicated port.
-func ForTenant(opts *serverOpts) {
-	opts.tenant = true
-}
-
 // WithInterceptor adds an additional interceptor. The interceptor is called before
 // streaming and unary RPCs and may inject an error.
-//
-// This option can only be used once (i.e. interceptors can not be chained).
 func WithInterceptor(f func(fullMethod string) error) ServerOption {
 	return func(opts *serverOpts) {
-		if opts.interceptor != nil {
-			panic("interceptor can only be set once")
+		if opts.interceptor == nil {
+			opts.interceptor = f
+		} else {
+			opts.interceptor = func(fullMethod string) error {
+				if err := opts.interceptor(fullMethod); err != nil {
```

Comment:
Seems to me that this will lead to infinite recursion as written. The new function will capture just the `opts`. The `interceptor` will be the function itself after the assignment.